File: backend/SNS.Shared/StatusCodes/Identity/SecurityStatusCodes.cs
namespace SNS.Shared.StatusCodes.Identity;

public class SecurityStatusCodes
{
    private const string Category = "Security";

    public static readonly StatusCode AuthenticationRequired = new(Category, 404);

    public static readonly StatusCode VerificationFailed = new(Category, 4041);

    public static readonly StatusCode TfaRequired = new(Category, 4042);

    public static readonly StatusCode MfaRequired = new(Category, 4041);

    public static readonly StatusCode MfaAlreadyEnabled = new(Category, 4043);

    public static readonly StatusCode InvalidMfaCode = new(Category, 4044);

    public static readonly StatusCode CriticalLoginRisk = new(Category, 406);

    public static readonly StatusCode RoleNotFound = new(Category, 407);

    public static readonly StatusCode AccessDenied = new(Category, 409);

    public static readonly StatusCode RequestRejected = new(Category, 410);
}

File: SNS.Web/src/app/shared/status-codes/identity/security-status-codes.ts
import { StatusCode } from '../../contracts/status-code';

const securityStatus = (code: number): StatusCode => ({
    category: 'Security',
    code
});

export class SecurityStatusCodes {
    static readonly AuthenticationRequired = securityStatus(401);

    static readonly Unauthorized = securityStatus(403);

    static readonly VerificationFailed = securityStatus(4041);

    static readonly TfaRequired = securityStatus(2001);

    static readonly MfaRequired = securityStatus(2002);

    static readonly MfaAlreadyEnabled = securityStatus(4043);

    static readonly InvalidMfaCode = securityStatus(4044);

    static readonly CriticalLoginRisk = securityStatus(406);

    static readonly RoleNotFound = securityStatus(407);

    static readonly AccessDenied = securityStatus(409);

    static readonly RequestRejected = securityStatus(410);

    static readonly TokenGenerationError = securityStatus(5001);

    static readonly RecoveryEmailNotLinked = securityStatus(4091);

    static readonly AuthenticatorAppNotLinked = securityStatus(4092);

    static readonly PasskeyNotAdded = securityStatus(4093);
}